gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientImpl.java

```diff
@@ -251,12 +257,43 @@ private static BlobWriteSessionConfig getSessionConfig(AsyncWriteChannelOptions
             writeOptions.getTemporaryPaths().stream()
                 .map(x -> Paths.get(x))
                 .collect(ImmutableSet.toImmutableSet()));
+      case PARALLEL_COMPOSITE_UPLOAD:
+        ExecutorService pcuThreadPool =
+            Executors.newCachedThreadPool(
+                new ThreadFactoryBuilder()
+                    .setNameFormat("gcsio-storage-client-pcu-pool-%d")
+                    .build());
+        return BlobWriteSessionConfigs.parallelCompositeUpload()
+            .withBufferAllocationStrategy(
+                BufferAllocationStrategy.fixedPool(
+                    writeOptions.getPCUBufferCount(), writeOptions.getPCUBufferCapacity()))
+            .withPartCleanupStrategy(getPartCleanupStrategy(writeOptions.getPartFileCleanupType()))
+            .withExecutorSupplier(ExecutorSupplier.useExecutor(pcuThreadPool))
```

Comment:
this should ideally be hidden from the client? What will happen if we do not provide the threadpool? i.e. will it create  one on its own?

Reply:
updated.